domene/src/main/kotlin/no/nav/tiltakspenger/saksbehandling/domene/vedtak/Vedtaksliste.kt

```diff
@@ -80,11 +84,35 @@ data class Vedtaksliste(
         }
     }
 
-    // TODO: Her må vi bruke periodiseringen fra behandlingen når den er klar
-    val tiltaksdeltagelseperioder: Periodisering<Tiltaksdeltagelse> by lazy {
+    private val tiltaksdeltakelsesperioderFraSaksopplysninger: Periodisering<Tiltaksdeltagelse> by lazy {
         saksopplysningerperiode.map { it.tiltaksdeltagelse.first() }
     }
 
+    // TODO: Blir dette riktig for revurdering og evt meldekort?
+    val valgteTiltaksdeltakelser: Periodisering<Tiltaksdeltagelse> by lazy {
+        innvilgetTidslinje.perioderMedVerdi.filter { it.verdi.behandling.valgteTiltaksdeltakelser != null }
+            .flatMap { it.verdi.behandling.valgteTiltaksdeltakelser!!.periodisering.krymp(it.periode).perioderMedVerdi }.let {
+                Periodisering(it)
+            }
+    }
+
+    fun valgteTiltaksdeltakelserForPeriode(periode: Periode): Periodisering<Tiltaksdeltagelse> {
+        return valgteTiltaksdeltakelser.krymp(periode)
```

Comment:
Her bør vi sikkert også endre til overlappendePeriode for å støtte meldekort utenfor innvilgelsesperioden